extras/logs/log_pull.py

```diff
@@ -0,0 +1,111 @@
+"""Retrieve log files to local folder for analysis.
+
+All log files are retrieved, as determined by the string '.log' in the file name
+Files are retrieved using SFTP.
+Further filtering of log types, date ranges, etc. should be performed on the local log copies as needed.
+
+"""
+import os
+from stat import S_ISDIR, S_ISREG
+import paramiko
+import argparse
+
+
+DESCRIPTION = "Securely retrieve log flies from a remove drive for analysis."
+
+def parse_arguments() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description=DESCRIPTION, formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    group = parser.add_argument_group(title='Device Discovery')
+    group.add_argument(
+        '--url',
+        default='neurodoor.nmr.mgh.harvard.edu',
+        type=str,
+        help='The address of the remote server holding the log files.'
+    )
+    group.add_argument(
+        '--user',
+        default=None,
+        required=True,
+        type=str,
+        help='The sftp user performing the transfer.'
+    )
+    group.add_argument(
+        '--pwd',
+        default=None,
+        required=True,
+        type=str,
+        help='The sftp password for the user performing the transfer.'
+    )
+    group.add_argument(
```

Comment:
If we want to support logs on drwho and neo, we could allow --remote-dir to be specified multiple times (e.g., using the `append` action) to cover all remote data directories. Or would it be easier to just make a wrapper script that calls the script N times with a different --remote-dir?

Reply:
(Because of the unified file system, we only need to specify one URL for either machine. Could even access via neurodoor.)